### ISSUE TITLE
What is the value of inference error?

### ISSUE THREAD

**Issue by wxyhv** [bytedance/lightseq#103]
Hello, I want to know if I use lightseq to do transformer model inference, the error value between lightseq with pytorch transformer in FP16.
 I don't find related document to explain about it.
 Thank you! looking for your reply~

**Comment by Taka152** [comment]
What is the error value? Could you show me some cases?

**Comment by wxyhv** [comment]
> What is the error value? Could you show me some cases?
"The error value" I said means the diff between "lightseq inference result" with "pytorch inference result" in transformer.
I aware that faster transformer inference result is different with pytorch or tensorflow, so I want to know the diff of lightseq infer result compare to pytorch infer result, especially in FP16 mode.

Thank you!

**Comment by wxyhv** [comment]
> What is the error value? Could you show me some cases?

What is the error value? Could you show me some cases?
"The error value" I said means the diff between "lightseq inference result" with "pytorch inference result" in transformer.
I aware that faster transformer inference result is different with pytorch or tensorflow, so I want to know the diff of lightseq infer result compare to pytorch infer result, especially in FP16 mode.

Thank you!

**Comment by neopro12** [comment]
The absolute difference between lightseq result and pytorch result is smaller than 1e-2, the relative difference  is smaller than 1e-3.

**Comment by wxyhv** [comment]
> The absolute difference between lightseq result and pytorch result is smaller than 1e-2, the relative difference is smaller than 1e-3.

wow, thank you! BTW, the difference you said is transformer FP32 mode or transformer FP16 mode? And Which op cause this difference?
Could you show a whole diffrence excel in the readme?

**Comment by neopro12** [comment]
FP16. Ops like layer norm and matrix multiplication may bring these difference. 
In general, because of the huge gap between top logits, the difference of logits between LS and pytorch will produce same result after softmax and topK.
